src/Middleware/UnauthorizedRedirectMiddleware.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Middleware;
+
+use Authentication\AuthenticationService;
+use Authentication\Authenticator\UnauthorizedException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Diactoros\Stream;
+
+/**
+ * Convert UnauthorizedExceptions into redirects.
+ */
+class UnauthorizedRedirectMiddleware
+{
+    /**
+     * The target URL to redirect to.
+     *
+     * @var string
+     */
+    protected $target;
+
+    /**
+     * Constructor
+     *
+     * @param string $target The target URL that redirects should be sent to.
+     */
+    public function __construct($target)
+    {
+        $this->target = $target;
+    }
+
+    /**
+     * Callable implementation for the middleware stack.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request.
+     * @param \Psr\Http\Message\ResponseInterface $response The response.
+     * @param callable $next The next middleware to call.
+     * @return \Psr\Http\Message\ResponseInterface A response.
+     */
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, $next)
+    {
+        try {
+            return $next($request, $response);
+        } catch (UnauthorizedException $e) {
+            return $response
+                ->withStatus(301)
+                ->withHeader('Location', $this->target);
```

Comment:
Should this allow cake array URLs as well, similar to how there's a UrlChecker for cake URLs? That would also let users have one login url that they can use throughout the various middleware.

I also noticed that the authorization middleware uses regular string urls, so there's a tad bit of inconsistency there. Perhaps we can add support for cake urls in the unauthorizedHandler too?

Reply:
The caller could use `Router::url()` before passing the URL into the middleware though.

Reply:
That is what I was thinking, but then why do we have `CakeRouterUrlChecker` to begin with? It's a tad confusing that within the same plugin some areas allow Cake array URLs and others don't.

If I had it my way, we wouldn't support them at all and just tell people to use the router. Would require less documentation too. But I think the inconsistency will lead to support issues.

Reply:
> That is what I was thinking, but then why do we have CakeRouterUrlChecker to begin with? 

I don't know. I didn't consider having people resolve their URLs before now.